backend/src/main/java/psi/controller/ChapterController.java

```diff
@@ -0,0 +1,58 @@
+package psi.controller;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+import psi.model.Chapter;
+import psi.model.Item;
+import psi.service.ChapterService;
+
+import java.util.List;
+
+@RestController
+@RequestMapping("/chapters")
+@CrossOrigin(origins = "http://localhost:4200")
+public class ChapterController {
+
+    @Autowired
+    private ChapterService chapterService;
+
+    @RequestMapping(value = "/", method = RequestMethod.GET)
+    public ResponseEntity<List<Chapter>> getChapters() {
+        return new ResponseEntity<>(chapterService.getChapters(),HttpStatus.OK);
+    }
+
+    @RequestMapping(value = "/{id}", method = RequestMethod.GET)
+    public ResponseEntity<Chapter> getChapter(@PathVariable String id) {
+        System.out.println("Calling controller wit id: " + id);
+        return new ResponseEntity<>(chapterService.findChapterById(id),HttpStatus.OK);
+    }
+
+    @RequestMapping(value = "/{id}/add-item", method = RequestMethod.POST)
+    public ResponseEntity<Chapter> addItemToChapter(@PathVariable String id, @RequestBody Item item) {
+        Chapter chapter = chapterService.findChapterById(id);
+        chapter.addItem(item);
```

Comment:
Not sure how 'find' works, i see that it throws a 'NoSuchElementException' - would that be inserted into an '404' response automatically? If not... maybe catch the exception and create the 404 response manually here? :D